iam.tf

```
@@ -1,20 +1,17 @@
-locals {
-  irsa_role_create = var.enabled && var.service_account_create && var.irsa_role_create
-}
-
 data "aws_iam_policy_document" "this" {
-  count = (local.irsa_role_create && var.irsa_policy_enabled) || (local.eks_pod_identity_role_create && var.eks_pod_identity_policy_enabled) ? 1 : 0
+  count = (var.enabled && (var.irsa_policy == null || var.pod_identity_policy == null)) ? 1 : 0
```

```suggestion
  count = var.enabled && ((var.irsa_policy_enabled != false && var.irsa_policy == null) || (var.pod_identity_policy_enabled != false && var.pod_identity_policy == null)) ? 1 : 0
```
I would add a check that use of a policy is not explicitly disabled

---

fixed.